plugins/modules/ome_application_certificate.py

```diff
@@ -160,37 +171,60 @@
 import json
 import os
 from ansible_collections.dellemc.openmanage.plugins.module_utils.ome import RestOME, OmeAnsibleModule
-from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from urllib.error import URLError, HTTPError
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible.module_utils.compat.version import LooseVersion
+
+OME_INFO = "ApplicationService/Info"
+LOW_OME = "OME version low"
 
 
 def get_resource_parameters(module):
     command = module.params["command"]
     csr_uri = "ApplicationService/Actions/ApplicationService.{0}"
     method = "POST"
+    payload = None
+
     if command == "generate_csr":
         uri = csr_uri.format("GenerateCSR")
-        payload = {"DistinguishedName": module.params["distinguished_name"],
-                   "DepartmentName": module.params["department_name"],
-                   "BusinessName": module.params["business_name"],
-                   "Locality": module.params["locality"], "State": module.params["country_state"],
-                   "Country": module.params["country"], "Email": module.params["email"],
-                   "San": get_san(module.params["subject_alternative_names"])}
-    else:
+        payload = {
+            "DistinguishedName": module.params["distinguished_name"],
+            "DepartmentName": module.params["department_name"],
+            "BusinessName": module.params["business_name"],
+            "Locality": module.params["locality"],
+            "State": module.params["country_state"],
+            "Country": module.params["country"],
+            "Email": module.params["email"],
+            "San": get_san(module.params["subject_alternative_names"])
+        }
+
+    elif command == "upload":
         file_path = module.params["upload_file"]
         uri = csr_uri.format("UploadCertificate")
         if os.path.exists(file_path):
-            with open(file_path, 'rb') as payload:
-                payload = payload.read()
+            with open(file_path, 'rb') as file:
+                payload = file.read()
+        else:
+            module.fail_json(msg="No such file or directory.")
+
+    elif command == "upload_cert_chain":
+        file_path = module.params["upload_file"]
+        uri = csr_uri.format("UploadCertChain")
+        if os.path.exists(file_path):
+            with open(file_path, 'rb') as file:
+                payload = file.read()
         else:
             module.fail_json(msg="No such file or directory.")
+
+    else:
```

Comment:
How will code reach invalid command when it is a choices.

Reply:
Done